website/scripts/build-static-content.js

```diff
@@ -17,59 +17,80 @@ module.exports = {
-        // Legacy:  (TODO: remove this later on, once in-progress frontend work is done)
-        await sails.helpers.compileMarkdownContent('handbook/queries/'); // TODO: pair w/ rachel and swap how page is built now that this part will work differently
-      },
-      async()=>{// Compile and generate XML sitemap
-        // TODO (see "refresh" action in sailsjs.com repo for example)
+      async()=>{// Parse markdown pages, compile & generate HTML files, and bake documentation's directory tree into the Sails app's configuration.
+
+        // Note:
+        // • path maths inspired by inspired by https://github.com/uncletammy/doc-templater/blob/2969726b598b39aa78648c5379e4d9503b65685e/lib/compile-markdown-tree-from-remote-git-repo.js#L107-L132
+        // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+        // // Original way that works:  (versus new stuff below)
+        // builtStaticContent.allPages = await sails.helpers.compileMarkdownContent('docs/');  // TODO remove this and helper once everything works again
+        // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+        builtStaticContent.allPages = [];// « dir tree representation that will be injected into Sails app's configuration
+
+        let SECTION_REPO_PATHS = ['docs/', 'handbook/'];
+        for (let sectionRepoPath of SECTION_REPO_PATHS) {
+          let thinTree = await sails.helpers.fs.ls.with({
+            dir: path.join(topLvlRepoPath, sectionRepoPath),
+            depth: 100,
+            includeDirs: false,
+            includeSymlinks: false,
+          });
+
+          for (let pageSourcePath of thinTree) {
+            let rootRelativeUrlPath = '/';// TODO: Perform path maths (determine this using sectionRepoPath, etc)
+            let lastModifiedAt = Date.now();// TODO: Check true modified date using git
+            let fallbackTitle = sails.helpers.strings.toSentenceCase(path.basename(pageSourcePath, '.ejs'));// « for clarity (the page isn't a template, necessarily, and this title is just a guess.  Display title will, more likely than not, come from a <docmeta> tag -- see the bottom of the original, raw unformatted markdown of any page in the sailsjs docs for an example of how to use docmeta tags)
+
+            // TODO: If markdown: Compile to HTML and parse docpage metadata
+
+            // TODO: Skip this page, if appropriate
+
+            // Generate HTML file
+            let htmlOutputPath = '';//TODO
+            if (dry) {
+              sails.log('Dry run: Would have generated file:', htmlOutputPath);
+            } else {
+              // TODO
+            }
+
+            // FUTURE: Figure out what to do about linked images (they'll get cached by CDN so probably ok to point at github, but markdown img srcs will break if relative.  Also GitHub could just change image URLs whenever.)
```

Comment:
@noahtalerman FYI this is probably gonna come up later, when we think we're almost done, and then none of the images in the documentation work